src/foam/core/menu/SessionLinkMenu.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * @license
+ * Copyright 2015 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.core.menu',
+  name: 'SessionLinkMenu',
+  extends: 'foam.core.menu.LinkMenu',
+
+  imports: [
+    'window'
+  ],
+
+  documentation: 'A menu item which links to an URL with current session id.',
+
+  methods: [
+    function select(X, menu) {
+      var url = window.location.origin;
+      url += this.link;
+      url += "?sessionId=" + localStorage.defaultSession;
```

Comment:
This wont work if the link already includes some params, which might be the case launching with a token or something, you can create a URI object before this and add params to it then convert it back to a string. Not sure if that is a major concern here though, could be left as a todo